kit/src/pool/mod.rs

```diff
@@ -16,6 +16,18 @@ pub trait PoolType {
 
     #[allow(async_fn_in_trait)]
     async fn swap_data(&self, pool_id: eU256, swap_x_in: bool, amount_in: eU256) -> Result<Bytes>;
+    /// Change Parameters
+    #[allow(async_fn_in_trait)]
+    async fn update_data(&self, new_data: Self::Parameters) -> Result<Bytes>;
+    /// Change Allocation Date
+    #[allow(async_fn_in_trait)]
+    async fn change_allocation_data(
+        &self,
+        pool_id: eU256,
+        is_allocate: bool,
+        amount_x: eU256,
+        amount_y: eU256,
+    ) -> Result<Bytes>;
```

Comment:
I think that this isn't quite the right signature for `change_allocation_data()` in general.

`ConstantSum` is weird. Since the pool only quotes a single price, you can allocate/deallocate any amount of `x` and `y`. In other pools such as `GeometricMean` or `LogNormal`, you can't. You have to instead provide an amount `x` OR `y` you want to allocate/deallocate and get the other amount from that. 

So of course this does work for `ConstantSum`, but I think the method should be made more flexible so it works for the other `PoolType`s. Thinking out loud here, we may almost want another associated type such as `AllocationData`. This could then be a tuple/struct for each pool. That is, we change to the signature:
```rust
async fn change_allocation_data(&self, pool_id: allocation_data: Self::AllocationData) -> Result<Bytes>
```
What do you think?

---

Also this is a nit, but I tend to prefer using an enum like
```rust
pub enum AllocateOrDeallocate {
    Allocate,
    Deallocate,
}
```
The reason why is for a Rust API, it is nice to know exactly what type you are referring to when you pass it in. A `bool` isn't really transparent with that. Basically, you get a type at compile time that is compiled into the same thing. Also, you can do pattern matching on the enum which is, in my opinion, a bit cleaner.  Perhaps you can change the `swap_data()` `bool` input to an enum as well (I know I wrote that, but I didn't think about this at the time)? We could add both these enum in the `pool/mod.rs`.

Reply:
I think these are good improvements, I was thinking about that with the constant sum but didn't think of the right way. I think the associated type is the way to go.